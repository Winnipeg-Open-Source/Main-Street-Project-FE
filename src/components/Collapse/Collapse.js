import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Collapse as Collapsable } from 'react-collapse';
import { Divider, Flex, Text } from 'pcln-design-system';
import { ChevronDown } from 'pcln-icons';
import Card from 'components/Card';
import useToggle from 'hooks/useToggle';

const ClickableFlex = styled(Flex)`
    cursor: pointer;
`;

const StyledFlex = styled(Flex)`
    & > :not(:last-child) {
        margin-bottom: 8px;
    }
`;

function Collapse ({ className, title, children }) {
    const [ isCollapsed, toggleCollapsed ] = useToggle();

    return (
        <Card className={className} isCollapsed={isCollapsed}>
            <ClickableFlex justifyContent='space-between' alignItems='center' onClick={toggleCollapsed}>
                <Text bold>{title}</Text>
                <ChevronDown />
            </ClickableFlex>
            <Collapsable isOpened={isCollapsed}>
                <StyledFlex flexDirection='column'>
                    <Divider />
                    {children}
                </StyledFlex>
            </Collapsable>
        </Card>
    );
}

Collapse.displayName = 'Collapse';

Collapse.propTypes = {
    className: PropTypes.string,
};

Collapse.defaultProps = {
    className: '',
};

export default memo(Collapse);
