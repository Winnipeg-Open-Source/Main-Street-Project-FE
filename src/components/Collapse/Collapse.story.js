import React from 'react';
import Collapse from './Collapse';

export const Default = () => (
    <Collapse title='Title'>
        <p>Children</p>
        <p>Children</p>
        <p>Children</p>
        <p>Children</p>
    </Collapse>
);

export default {
    title: 'Collapse',
    component: Collapse,
};
