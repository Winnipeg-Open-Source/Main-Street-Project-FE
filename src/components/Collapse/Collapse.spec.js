import React from 'react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import Collapse from './Collapse';

describe ('Collapse', () => {
    it ('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<Collapse />);
        expect(asFragment()).toMatchSnapshot();
    });
});
