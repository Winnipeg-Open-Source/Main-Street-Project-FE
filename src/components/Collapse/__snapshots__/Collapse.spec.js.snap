// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <div
      class="sc-AxirZ sc-pQfvp sc-pAMRk djnmSu"
      color="background.lightest"
      width="1"
    >
      <div
        class="sc-AxirZ sc-AxhCb sc-pIITJ JHjYq"
      >
        <div
          class="sc-AxhUy gGlGcm"
        />
        <svg
          aria-hidden="true"
          class="sc-AxmLO bliMTQ sc-fzqAui eslvzV"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
          />
        </svg>
      </div>
      <div
        class="ReactCollapse--collapse"
        style="height: 0px; overflow: hidden;"
      >
        <div
          class="ReactCollapse--content"
        >
          <div
            class="sc-AxirZ sc-AxhCb sc-pRFZy cxlcIV"
          >
            <hr
              class="sc-oUAoT bfCxJq"
              color="border"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
